src/infra/core/src/pull_service_impl.rs

```diff
@@ -257,9 +265,14 @@ impl PullServiceImpl {
             }
         }
 
-        // No luck - now have to search through aliases
+        // No luck - now have to search through aliases (of current user)
         use tokio_stream::StreamExt;
-        let mut datasets = self.local_repo.get_all_datasets();
+        let mut datasets = if self.dataset_repo.is_multi_tenant() {
```

Comment:
Same here, if `current_account.account_name` is `kamu` in single-tenant mode then two branches should return same result?

Reply:
Right now a single-tenant dataset repository would panic if we query it by account:

```
    fn get_account_datasets<'s>(&'s self, _account_name: AccountName) -> DatasetHandleStream<'s> {
        panic!("Single-tenant dataset repository queried by account");
    }
```

What you basically suggest could work if we compared the account name with `DEFAULT_DATASET_OWNER_NAME` here and redirect to `get_all_datasets`. Still panicking if name does not match default.

Reply:
Yes, that's exactly what I had in mind. Making single-tenant repo work:
- with aliases without `<account>`
- and aliases where account matches the one in `CurrentAccountConfig`

Reply:
Improved